go/lib/layers/extensions_layer.go

```diff
@@ -84,6 +86,22 @@ func (e *Extension) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) err
 }
 
 func (e *Extension) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
+	bytes, err := e.serialize(b, opts)
+	if err != nil {
+		return err
+	}
+
+	//The bytes that need to be authenticated
+	switch e.Class {
```

Comment:
why don't we have different behaviors for these? Interface `Extension` implemented by a base class and e2e and hbh.
Or, since the caller will know if they are e2e or hbh, let them pass that information along.